documentation/docs/examples/cad-3dobjects.mdx

```diff
@@ -0,0 +1,120 @@
+---
+id: cad-3dobjects
+title: Add custom 3D objects
+description: Extend the scene with custom ThreeJS objects.
+---
+
+import { Water } from 'three/examples/jsm/objects/Water';
+import { LiveCodeSnippet }from '../../src/components/LiveCodeSnippet';
+import { DemoWrapper } from '../../src/components/DemoWrapper';
+import { Cognite3DModel, Cognite3DViewer } from '@cognite/reveal';
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Sometimes it's useful to add custom content to the 3D scene to model data Reveal doesn't support or to customize the 3D environment. Reveal supports this by using the `Cognite3DViewer.addObject3D`-function which accepts [ThreeJS](https://threejs.org/) objects. 
+
+:::tip
+Reveal uses an "implicit lighting model", i.e. no lights are added to the scene. If you need lights this needs to be added to the viewer scene. 
+:::
+
+<DemoWrapper modelId={4715379429968321} revisionId={5688854005909501} name="Cognite3DViewerDemo" />
+
+## Adding markers to the scene on clicks
+
+The following examples reacts to clicks in the scene and adds markers to the positions clicked.
+
+<LiveCodeSnippet>
+{`
+const markerGeometry = new THREE.SphereBufferGeometry(1, 10, 10);
+const markerMaterial = new THREE.MeshStandardMaterial({ emissive: 'white', wireframe: true });
+viewer.on('click', (event) => {
+  const intersection = viewer.getIntersectionFromPixel(event.offsetX, event.offsetY);
+  if (intersection) {
+    const marker = new THREE.Mesh(markerGeometry, markerMaterial);
+    marker.position.copy(intersection.point);
+    viewer.addObject3D(marker);
+  };
+});
+`}
+</LiveCodeSnippet>
+
+## Add environment
+
+To make a model feel more natural it might be useful to add an environment. The following examples adds sky and a sea effect for the example oil rig. 
+
+First, lets encapsulate the model in a large sky-textured sphere to create an illusion of skies. The sphere map is available from [Free HDR - EXR SKIES-blog](http://freepanorama.blogspot.com/).
+
+<LiveCodeSnippet scope={{skyUrl: useBaseUrl('/img/sky007.jpg')}}>
```

Comment:
That's just an opinion and you can ignore it if you want, but I think not every example must be interactive. I think it's even better if you have a complete demo in the beginning or at the end of the page with some static code snippets that show how to achieve that state. It's just might not be obvious (or convenient) to a user to click "run" on every code snippet to get the cool final state with the sky and sea.  

This article looks like a tutorial, so perhaps it's better to just illustrate each step with a picture rather than expect that a user will click "run" and play with the model after each step.

Reply:
I've split into static + interactive example